File: StriverA2Z/Java/LinkedLists/Questions/DoublyLinkedList/BrowserHistory.java
public class BrowserHistory {
    Node currentPage;

    BrowserHistory(int homepage) {
        currentPage = new Node(homepage);
    }

    void visit(int url) {
        Node newNode = new Node(url);
        newNode.prev = currentPage;
        currentPage.next = newNode;
        currentPage = newNode;
    }

    int forward(int steps) {
        while (steps != 0) {
            if (currentPage.next != null) currentPage = currentPage.next;
            else break;
            steps--;
        }
        return currentPage.data;
    }

    int back(int steps) {
        while (steps != 0) {
            if (currentPage.prev != null) currentPage = currentPage.prev;
            else break;
            steps--;
        }
        return currentPage.data;
    }
}

File: StriverA2Z/JavaScript&TypeScript/src/LinkedLists/Questions/DLL/browserHistory.ts
import {DoubleLinkNode} from "../../DoublyLinkedList/doublyLinkedList";

class Browser {
	currentPage: DoubleLinkNode;

	constructor(homepage: number) {
		this.currentPage = new DoubleLinkNode(homepage);
	}

	visit(url: number) {
		let newNode = new DoubleLinkNode(url);
		newNode.prev = this.currentPage;
		this.currentPage.next = newNode;
		this.currentPage = newNode;
	}

	forward(steps: number) {
		while (steps) {
			if (this.currentPage.next) this.currentPage = this.currentPage.next; else break;
			steps--;
		}
		return this.currentPage.data;
	}

	back(steps: number) {
		while (steps) {
			if (this.currentPage.prev) this.currentPage = this.currentPage.prev; else break;
			steps--;
		}
		return this.currentPage.data;
	}
}